### ISSUE TITLE
Error during rewards withdrawal 

### ISSUE THREAD

**Issue by asadovka** [cybercongress/cyberd#302]
### Current Behavior
[comment]: # (Describe what actually happened.)

I ran this command:

```

docker run -ti --rm -v /Users/siarheisadau/cyberd/cyberdcli:/root/.cyberdcli cyberd/cyberd:euler-4 cyberdcli tx distr withdraw-all-rewards \
--from=asadovka \
--chain-id=euler-4 \
--node=86.57.254.202:36657

```
and get this error:

```
Response:
  Height: 116101
  TxHash: 6F8054F1A5AB07AE6548AFD571E115915A0A7618762FA1DFFE3F92ACC8DC9CA8
  Code: 1
  GasUsed: 73521
  Codespace: sdk
ERROR: {"codespace":"sdk","code":1,"message":"recovered: calculated final stake for delegator cyber14d92r4svhl4qa3g6q48tjekarw2kt67nvcqwg7 greater than current stake: 500000000.000000000000000000, 499500000.000000000000000000\nstack:\ngoroutine 264798 [running]:\nruntime/debug.Stack(0xc01ad9c4b0, 0xf89ac0, 0xc053f7f640)\n\t/usr/local/go/src/runtime/debug/stack.go:24 +0x9d\ngithub.com/cosmos/cosmos-sdk/baseapp.(*BaseApp).runTx.func1(0xc01ad9d490, 0xc01ad9d4b8, 0xc01ad9d5f0)\n\t/root/go/pkg/mod/github.com/cosmos/cosmos-sdk@v0.33.0/baseapp/baseapp.go:757 +0x90\npanic(0xf89ac0, 0xc053f7f640)\n\t/usr/local/go/src/runtime/panic.go:522 +0x1b5\ngithub.com/cosmos/cosmos-sdk/x/distribution/keeper.Keeper.calculateDelegationRewards(0x1425be0, 0xc0009553a0, 0xc0004d04d0, 0xc0004d04d0, 0x1425be0, 0xc0009553e0, 0x1425c20, 0xc0009553f0, 0xc0000d4bc0, 0x5, ...)\n\t/root/go/pkg/mod/github.com/cosmos/cosmos-sdk@v0.33.0/x/distribution/keeper/delegation.go:95 +0x743\ngithub.com/cosmos/cosmos-sdk/x/distribution/keeper.Keeper.withdrawDelegationRewards(0x1425be0, 0xc0009553a0, 0xc0004d04d0, 0xc0004d04d0, 0x1425be0, 0xc0009553e0, 0x1425c20, 0xc0009553f0, 0xc0000d4bc0, 0x5, ...)\n\t/root/go/pkg/mod/github.com/cosmos/cosmos-sdk@v0.33.0/x/distribution/keeper/delegation.go:114 +0x301\ngithub.com/cosmos/cosmos-sdk/x/distribution/keeper.Keeper.WithdrawDelegationRewards(0x1425be0, 0xc0009553a0, 0xc0004d04d0, 0xc0004d04d0, 0x1425be0, 0xc0009553e0, 0x1425c20, 0xc0009553f0, 0xc0000d4bc0, 0x5, ...)\n\t/root/go/pkg/mod/github.com/cosmos/cosmos-sdk@v0.33.0/x/distribution/keeper/keeper.go:62 +0x20c\ngithub.com/cosmos/cosmos-sdk/x/distribution.handleMsgWithdrawDelegatorReward(0x1434ba0, 0xc044e28ae0, 0xc00e77ca80, 0x14, 0xc0351be340, 0x14, 0x14, 0xc0351be360, 0x14, 0x14, ...)\n\t/root/go/pkg/mod/github.com/cosmos/cosmos-sdk@v0.33.0/x/distribution/handler.go:45 +0x123\ngithub.com/cosmos/cosmos-sdk/x/distribution.NewHandler.func1(0x1434ba0, 0xc044e28ae0, 0xc00e77ca80, 0x14, 0x1438ae0, 0xc0765e2960, 0x0, 0x0, 0x0, 0x0, ...)\n\t/root/go/pkg/mod/github.com/cosmos/cosmos-sdk@v0.33.0/x/distribution/handler.go:17 +0x2f7\ngithub.com/cosmos/cosmos-sdk/baseapp.(*BaseApp).runMsgs(0xc000be5c00, 0x1434ba0, 0xc044e28ae0, 0xc00e77ca80, 0x14, 0xc00e77dd40, 0x4, 0x4, 0x1434b02, 0x0, ...)\n\t/root/go/pkg/mod/github.com/cosmos/cosmos-sdk@v0.33.0/baseapp/baseapp.go:653 +0xb4e\ngithub.com/cosmos/cosmos-sdk/baseapp.(*BaseApp).runTx(0xc000be5c00, 0x102, 0xc056a05040, 0x140, 0x140, 0x1427da0, 0xc0194e94a0, 0x0, 0x0, 0x0, ...)\n\t/root/go/pkg/mod/github.com/cosmos/cosmos-sdk@v0.33.0/baseapp/baseapp.go:830 +0x375\ngithub.com/cosmos/cosmos-sdk/baseapp.(*BaseApp).DeliverTx(0xc000be5c00, 0xc056a05040, 0x140, 0x140, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)\n\t/root/go/pkg/mod/github.com/cosmos/cosmos-sdk@v0.33.0/baseapp/baseapp.go:587 +0x2a0\ngithub.com/cybercongress/cyberd/app.(*CyberdApp).DeliverTx(0xc000120b00, 0xc056a05040, 0x140, 0x140, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)\n\t/sources/app/app.go:381 +0x527\ngithub.com/tendermint/tendermint/abci/client.(*localClient).DeliverTxAsync(0xc00008b740, 0xc056a05040, 0x140, 0x140, 0x0)\n\t/root/go/pkg/mod/github.com/tendermint/tendermint@v0.31.0-dev0/abci/client/local_client.go:88 +0xc5\ngithub.com/tendermint/tendermint/proxy.(*appConnConsensus).DeliverTxAsync(0xc000954180, 0xc056a05040, 0x140, 0x140, 0xa)\n\t/root/go/pkg/mod/github.com/tendermint/tendermint@v0.31.0-dev0/proxy/app_conn.go:73 +0x51\ngithub.com/tendermint/tendermint/state.execBlockOnProxyApp(0x14356e0, 0xc00269c960, 0x143e8c0, 0xc000954180, 0xc0058db4a0, 0xc00bed69c0, 0x1446dc0, 0xc0000101e8, 0xc00fa61780, 0x7, ...)\n\t/root/go/pkg/mod/github.com/tendermint/tendermint@v0.31.0-dev0/state/execution.go:281 +0x5e2\ngithub.com/tendermint/tendermint/state.(*BlockExecutor).ApplyBlock(0xc0000fd110, 0xa, 0x0, 0xc00fa61770, 0x6, 0xc00fa61780, 0x7, 0x1c584, 0x84, 0xc037356980, ...)\n\t/root/go/pkg/mod/github.com/tendermint/tendermint@v0.31.0-dev0/state/execution.go:119 +0x1b5\ngithub.com/tendermint/tendermint/consensus.(*ConsensusState).finalizeCommit(0xc00048fc00, 0x1c585)\n\t/root/go/pkg/mod/github.com/tendermint/tendermint@v0.31.0-dev0/consensus/state.go:1327 +0x9be\ngithub.com/tendermint/tendermint/consensus.(*ConsensusState).tryFinalizeCommit(0xc00048fc00, 0x1c585)\n\t/root/go/pkg/mod/github.com/tendermint/tendermint@v0.31.0-dev0/consensus/state.go:1258 +0x3e0\ngithub.com/tendermint/tendermint/consensus.(*ConsensusState).enterCommit.func1(0xc00048fc00, 0x0, 0x1c585)\n\t/root/go/pkg/mod/github.com/tendermint/tendermint@v0.31.0-dev0/consensus/state.go:1204 +0x91\ngithub.com/tendermint/tendermint/consensus.(*ConsensusState).enterCommit(0xc00048fc00, 0x1c585, 0x0)\n\t/root/go/pkg/mod/github.com/tendermint/tendermint@v0.31.0-dev0/consensus/state.go:1235 +0x61e\ngithub.com/tendermint/tendermint/consensus.(*ConsensusState).addVote(0xc00048fc00, 0xc0318832c0, 0xc00283f380, 0x28, 0x109, 0x109, 0x109)\n\t/root/go/pkg/mod/github.com/tendermint/tendermint@v0.31.0-dev0/consensus/state.go:1660 +0xaec\ngithub.com/tendermint/tendermint/consensus.(*ConsensusState).tryAddVote(0xc00048fc00, 0xc0318832c0, 0xc00283f380, 0x28, 0x0, 0x109, 0x101)\n\t/root/go/pkg/mod/github.com/tendermint/tendermint@v0.31.0-dev0/consensus/state.go:1506 +0x5c\ngithub.com/tendermint/tendermint/consensus.(*ConsensusState).handleMsg(0xc00048fc00, 0x1417f60, 0xc02b00da38, 0xc00283f380, 0x28)\n\t/root/go/pkg/mod/github.com/tendermint/tendermint@v0.31.0-dev0/consensus/state.go:694 +0x6cc\ngithub.com/tendermint/tendermint/consensus.(*ConsensusState).receiveRoutine(0xc00048fc00, 0x0)\n\t/root/go/pkg/mod/github.com/tendermint/tendermint@v0.31.0-dev0/consensus/state.go:642 +0x695\ncreated by github.com/tendermint/tendermint/consensus.(*ConsensusState).OnStart\n\t/root/go/pkg/mod/github.com/tendermint/tendermint@v0.31.0-dev0/consensus/state.go:334 +0x13a\n"}
asadovka:~ siarheisadau$ docker run -ti --rm -v /Users/siarheisadau/cyberd/cyberdcli:/root/.cyberdcli cyberd/cyberd:euler-4 cyberdcli  tx distr withdraw-rewards cybervaloper18wn5kyxewwk6qh97493xtj9gufpmqserdykng9
```
Problem is with cyb_ba validator (cybervaloper18z9vsjffvggluxcxfqwsmrz02v37hx2e4ya023). He has been jailed for 20 minutes and lost some of his and mine stake.

### Expected Behavior
[comment]: # (Describe what you expected to happen.)

I expected withdrawal of all rewards.

**Comment by asadovka** [comment]
got this bug with every jailed validator

**Comment by SaveTheAles** [comment]
Same things with unbond command

**Comment by litvintech** [comment]
@hleb-albau please, mention here your issue from cosmos-sdk

**Comment by litvintech** [comment]
referenced to
https://github.com/cosmos/cosmos-sdk/issues/4012

**Comment by hleb-albau** [comment]
Fixed for 0.1.5 release.
